src/content/app/genome-browser/state/browser-entity/browserEntitySlice.ts

```diff
@@ -0,0 +1,219 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Action,
+  ActionCreator,
+  createSlice,
+  PayloadAction,
+  ThunkAction
+} from '@reduxjs/toolkit';
+import pickBy from 'lodash/pickBy';
+import merge from 'lodash/merge';
+
+import browserStorageService from 'src/content/app/genome-browser/services/browser-storage-service';
+import trackPanelStorageService from 'src/content/app/genome-browser/components/track-panel/services/track-panel-storage-service';
+
+import { fetchEnsObject } from 'src/shared/state/ens-object/ensObjectActions';
+import {
+  ChrLocation,
+  deleteGenomeLocationData,
+  setLocationDataFromUrl
+} from 'src/content/app/genome-browser/state/browser-location/browserLocationSlice';
+import {
+  deleteGenomeTrackPanelData,
+  setTrackPanelDataFromUrl
+} from 'src/content/app/genome-browser/state/track-panel/trackPanelSlice';
+import {
+  getBrowserActiveEnsObjectIds,
+  getBrowserTrackStates,
+  getBrowserActiveGenomeId
+} from 'src/content/app/genome-browser/state/browser-entity/browserEntitySelectors';
+import { ensureSpeciesIsEnabled } from 'src/content/app/species-selector/state/speciesSelectorSlice';
+
+import {
+  BrowserTrackStates,
+  TrackActivityStatus
+} from 'src/content/app/genome-browser/components/track-panel/trackPanelConfig';
+import { RootState } from 'src/store';
+
+export type BrowserEntityState = Readonly<{
+  activeGenomeId: string | null;
+  activeEnsObjectIds: { [genomeId: string]: string };
+  trackStates: BrowserTrackStates;
+}>;
+
+const isServer = typeof window === 'undefined';
+
+const activeGenomeId = !isServer
+  ? browserStorageService.getActiveGenomeId()
+  : null;
+const activeEnsObjectIds = !isServer
+  ? browserStorageService.getActiveEnsObjectIds()
+  : [];
+const trackStates = !isServer ? browserStorageService.getTrackStates() : {};
```

Comment:
Would it work if we initialise the state with the server-side values, and then dispatch a dedicated action from [here](https://github.com/Ensembl/ensembl-client/blob/dev/src/root/useRestoredReduxState.ts#L26) to update the state to client-side?

Reply:
That's done